src/Raven.Server/ServerWide/Maintenance/ClusterObserver.cs

```diff
@@ -397,6 +405,62 @@ internal async Task CleanUpUnusedAutoIndexes(DatabaseObservationState databaseSt
             }
         }
 
+        internal async Task CleanUpCompareExchangeTombstones()
+        {
+            using (_server.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            using (context.OpenReadTransaction())
+            {
+                foreach (var dbName in _server.Cluster.GetDatabaseNames(context))
+                {
+                    if (_server.Cluster.GetNumberOfCompareExchangeTombstones(context, dbName) > 0)
+                    {
+                        var maxEtag = GetMaxTombstonesEtagToDelete(context, dbName);
+                        if (maxEtag > 0)
+                        {
+                            var result = await _server.SendToLeaderAsync(new CleanCompareExchangeTombstonesCommand(dbName, maxEtag));
```

Comment:
Let's move this all outside of the open transaction

Reply:
done